core/src/test/java/org/elasticsearch/index/get/GetResultTests.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.get;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.common.xcontent.XContentHelper.toXContent;
+import static org.elasticsearch.index.get.GetFieldTests.assertSameOutput;
+import static org.elasticsearch.index.get.GetFieldTests.randomGetField;
+import static org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode;
+
+public class GetResultTests extends ESTestCase {
+
+    public void testToAndFromXContent() throws Exception {
+        XContentType xContentType = randomFrom(XContentType.values());
+        Tuple<GetResult, GetResult> tuple = randomGetResult(xContentType);
+        GetResult getResult = tuple.v1();
+        GetResult expectedGetResult = tuple.v2();
+        BytesReference originalBytes = toXContent(getResult, xContentType, false);
+        //test that we can parse what we print out
+        GetResult parsedGetResult;
+        try (XContentParser parser = createParser(xContentType.xContent(), originalBytes)) {
+            parsedGetResult = GetResult.fromXContent(parser);
+            assertNull(parser.nextToken());
+        }
+        assertEquals(expectedGetResult, parsedGetResult);
+        //print the parsed object out and test that the output is the same as the original output
+        BytesReference finalBytes = toXContent(parsedGetResult, xContentType, false);
+        assertSameOutput(originalBytes, finalBytes, xContentType);
```

Comment:
Can we add a comment line like:
```
// GetResult can contain stored fields of various types, including floats, where order cannot be guaranteed. 
// Since a byte-to-byte comparison cannot be done we use assertSameOutput.
```